extern/iso_c_fortran_bmi/README.md

```diff
@@ -0,0 +1,100 @@
+# ISO C Fortran BMI bindings
+
+## About
+
+This is a small library that exposes a C/Fortran interoperability layer for BMI functions.  The library presents the BMI functions as a set of free functions that operate on a properly initialized and boxed Fortran BMI object.  This requires Fortran BMI models to implement a `register_bmi` function which, when called by a C/C++ executable, provides an opaque handle to the underlying BMI type `BMI_TYPE`:
+
+```fortran
+function register_bmi(this) result(bmi_status) bind(C, name="register_bmi")
+   use, intrinsic:: iso_c_binding, only: c_ptr, c_loc, c_int
+   use iso_c_bmif_2_0
+   implicit none
+   type(c_ptr) :: this
+   integer(kind=c_int) :: bmi_status
+   !Create the model instance to use
+   type(BMI_TYPE), target, save :: bmi_model !need to ensure scope/lifetime, use save attribute
```

Comment:
Is this still accurate> I remember you saying that you removed the dependency on 'save'

Reply:
I just pushed an update to the register function, I'll fix this documentation real quick.

Reply:
updated with a0ca7e9fe21cd5c120d2be04331d6e6c23b62796